Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,52 +2,88 @@
+ * @param performMultiLevelLookup Set the `DOTNET_MULTILEVEL_LOOKUP`environment variable
+ */
+async function installDotNet(
+    installationPath: string,
+    packageType: string,
+    versionSpec: string | null,
+    useGlobalJson: boolean,
+    workingDirectory: string | null,
+    includePreviewVersions: boolean,
+    performMultiLevelLookup: boolean) {
 
-        let versionFetcher = new DotNetCoreVersionFetcher();
+    if (!installationPath || installationPath.length == 0) {
+        installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
+    }
+    let versionFetcher = new DotNetCoreVersionFetcher();
+    let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
+    // here we must check also the package type because if the user switch the packageType the useGlobalJson can be true, also if it will hidden.
+    if (useGlobalJson && packageType == "sdk") {
+        let globalJsonFetcherInstance = new globalJsonFetcher(workingDirectory);
+        let versionsToInstall: VersionInfo[] = await globalJsonFetcherInstance.GetVersions(packageType);
+        versionsToInstall = versionsToInstall.filter(d => !dotNetCoreInstaller.isVersionInstalled(d.getVersion()));
+        for (let index = 0; index < versionsToInstall.length; index++) {
+            const version = versionsToInstall[index];
+            let url = versionFetcher.getDownloadUrl(version);
+            if (!dotNetCoreInstaller.isVersionInstalled(version.getVersion())) {
+                await dotNetCoreInstaller.downloadAndInstall(version, url);
+            }
+        }
+    } else if (versionSpec) {
+        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+        let versionSpecParts = new VersionParts(versionSpec);
         let versionInfo: VersionInfo = await versionFetcher.getVersionInfo(versionSpecParts.versionSpec, packageType, includePreviewVersions);
         if (!versionInfo) {
             throw tl.loc("MatchingVersionNotFound", versionSpecParts.versionSpec);
         }
-
-        let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
         if (!dotNetCoreInstaller.isVersionInstalled(versionInfo.getVersion())) {
             await dotNetCoreInstaller.downloadAndInstall(versionInfo, versionFetcher.getDownloadUrl(versionInfo));
         }
-
-        tl.prependPath(installationPath);
-
-        // Set DOTNET_ROOT for dotnet core Apphost to find runtime since it is installed to a non well-known location.
-        tl.setVariable('DOTNET_ROOT', installationPath);
-
-        // By default disable Multi Level Lookup unless user wants it enabled.
-        let performMultiLevelLookup = tl.getBoolInput("performMultiLevelLookup", false);
-        tl.setVariable("DOTNET_MULTILEVEL_LOOKUP", !performMultiLevelLookup ? "0" : "1");
+    } else {
+        throw new error("Hey developer you have called the method `installDotNet` without a `versionSpec` or without `useGlobalJson`. that is impossible.");
```

Comment:
Add this message to task.json and throw using tl.loc

Reply:
No I have removed this because this is a only developer message. That's only be happened if the method 'install Dotnet' is called wrong. In a normal language I would overload this method but that isn't supported by typescript. So I need errors 🤷‍♀️

Reply:
I have removed the link to the task.json yesterday after a comment of @sachinma. Please let us discuss this here together to avoid build it in and out every day :)